### ISSUE TITLE
PEN-1000.3.5 As a pool owner/user I need to be able to within the installation define capabilities of heaters assoc. with bodies of water so I can manage my pool/spa (temps not saving correctly)

### ISSUE THREAD

**Issue by acarnagey** [gryz/THW-Update-Needed#20]
When I try to create or edit heaters with params START and STOP, I get different values then when I set the  params.  Tested on DEV2.

Sending: '{"command":"CREATEOBJECT","objectList":[{"objtyp":"HEATER","params":{"SNAME":"g2","SUBTYP":"SOLAR","BODY":"BF8A3","START":2,"STOP":4,"COOL":"OFF"}}],"messageID":"20317a51-38ef-46ed-a9bd-b730e7cdded3"}' : Mon Aug 01 2016 12:33:07 GMT-0500 (CDT)

SocketService.js:91 Receiving: {"objnam":"H2DCB","command":"ObjectCreated","messageID":"20317a51-38ef-46ed-a9bd-b730e7cdded3","response":"201"}: Mon Aug 01 2016 12:33:07 GMT-0500 (CDT)

SocketService.js:34 Sending: '{"command":"RequestParamList","objectList":[{"objnam":"H2DCB","keys":["OBJTYP","SNAME","STATUS","SUBTYP","PERMIT","TIMOUT","READY","HTMODE","SHOMNU","COOL","COMUART","BODY","HNAME","START","STOP","HEATING"]}],"messageID":"b318ccf7-ad7d-4cd8-8103-4b1fd07f40a9"}' : Mon Aug 01 2016 12:33:07 GMT-0500 (CDT)

SocketService.js:91 Receiving: {"objectList":[{"objnam":"H2DCB","params":{"OBJTYP":"HEATER","SNAME":"g2","STATUS":"OFF","SUBTYP":"SOLAR","PERMIT":"PERMIT","TIMOUT":"TIMOUT","READY":"READY","HTMODE":"SOLAR","SHOMNU":"hrtco","COOL":"OFF","COMUART":"COMUART","BODY":"BF8A3","HNAME":"Solar Heater 1","START":"6","STOP":"3","HEATING":"OFF"}}],"command":"NotifyList","messageID":"b318ccf7-ad7d-4cd8-8103-4b1fd07f40a9","response":"200","timeNow":"949539803","timeSince":"949539636"}

**Comment by edgardoatpentair** [comment]
YOU HAVE TO INCLUDE THE "" QUOTATION MARK FOR EVERY KEY-PAIR VALUE 
EACH ONE MUST BE FORMATTED AS STRING., NOTICE "2" AND "5" BELOW
I CHECK AND VALIDATE THIS ISSUE CHECK ATTACHED REPORT.
[createHeater.txt](https://github.com/gryz/THW-Update-Needed/files/397208/createHeater.txt)

**Comment by acarnagey** [comment]
Even with the quotations the values for start and stop in NotifyList (5 and 3) do not match the existing CreateObject (2 and 5).  See in the createHeater.txt you have attached.

**Comment by edgardoatpentair** [comment]
Yes I notice that ,thank you the values are converted from F to C, I'm fixing this issue on the app module solar Heaters

**Comment by edgardoatpentair** [comment]
the issue I found is on the GUI., so not affect your web interface. so after further investigation, I found there are rules to set start delta and stop delta. here is the document , but in resume
1- set difference (start and stop) minimum to 2
2- set start is always greater than stop 
3- the ranges are :
Start Temp. Delta:  6° (default): 4°F - 9°F adjustable. The minimum temperature difference to start heating.
Stop Temp. Delta:  3° (default): 2°F - 5°F adjustable: The temperature difference at which Heating will stop. below is full description about it

**Comment by edgardoatpentair** [comment]
Solar: Heater type. Enter a new heater name (optional), affiliated body of water (pool), this name and heater type are defined in the Body Configuration > Pool Settings.
Start Temp. Delta:  6° (default): 4°F - 9°F adjustable. The minimum temperature difference to start heating.
Stop Temp. Delta:  3° (default): 2°F - 5°F adjustable: The temperature difference at which Heating will stop.
Note: Effective solar heat depends on the solar and water temperature and the start and stop
Differentials (Delta), When solar is available, then the IntellliCenter Controller will rotate the positive 3-way valve and turn on the solar booster pump if applicable or increase the RPM’s if an IntelliFlo® or IntelliPro® Pump is present. This allows the water to flow to the solar collectors. If a pressure cleaner pump is installed, the cleaner pump will turn off for 10 minutes every time the IntelliCenter Controller turns ON. The solar roof sensor must be 6° higher than the water temperature. The pump must be ON for a few minutes for operation. When the roof sensor and the water temperature are within 3° of each other, solar heat will switch off. The solar temperature start and stop differential settings are factory set at 6° and 3°.
 
Nocturnal Cooling ON/OFF: Provides automatic nocturnal (nighttime) cooling during night time hours for pools that overheat in hot climates. When it enabled, relatively warmer pool water circulates to the collector panels when the collector temperature and the pool temperature is hotter than the high limit temperature setting. This process cools the pool water over time. Circulation will stop when the collector temperature is 3° (default) less than the pool temperature or the pool temperature is cooler than the high limit setting.

**Comment by edgardoatpentair** [comment]
here is the test  result,  I created for you to v
[createHeater.txt](https://github.com/gryz/THW-Update-Needed/files/397822/createHeater.txt)
alidate the parameters , no change on the App so you should be able to set values with the rules above described.

**Comment by acarnagey** [comment]
Ok, I'll take a look at this soon, thanks.

**Comment by acarnagey** [comment]
This is not working again.  Tested on Dev1.

Sending: '{"command":"SETPARAMLIST","objectList":[{"objnam":"H66AE","params":{"SNAME":"S2","SUBTYP":"SOLAR","START":"4","STOP":"2","COOL":"ON"}}],"messageID":"9e2ae137-a9c7-4b8b-9fc5-ad6e566d0b0f"}' : Mon Aug 15 2016 11:26:04 GMT-0500 (CDT)

SocketService.js:91 Receiving: {
    "command": "NotifyList",
    "messageID": "0c891fa1-4045-407b-9829-468028effbb3",
    "objectList": [
        {
            "objnam": "H66AE",
            "params": {
                "START": "5"
            }
        }
    ]
}: Mon Aug 15 2016 11:26:05 GMT-0500 (CDT)

SocketService.js:91 Receiving: {
    "command": "NotifyList",
    "messageID": "b67ad37b-50db-4fcd-9d3f-dc3bf61f1bf4",
    "objectList": [
        {
            "objnam": "H66AE",
            "params": {
                "STOP": "2"
            }
        }
    ]
}: Mon Aug 15 2016 11:26:05 GMT-0500 (CDT)

SocketService.js:91 Receiving: {"command":"SetParamList","messageID":"9e2ae137-a9c7-4b8b-9fc5-ad6e566d0b0f","response":"200"}: Mon Aug 15 2016 11:26:05 GMT-0500 (CDT)
SocketService.js:34 Sending: '{"command":"RequestParamList","objectList":[{"objnam":"H66AE","keys":["OBJTYP","SNAME","STATUS","SUBTYP","PERMIT","TIMOUT","READY","HTMODE","SHOMNU","COOL","COMUART","BODY","HNAME","START","STOP","HEATING"]}],"messageID":"79e279d7-5d5e-4ecd-b129-8601c5ecca9d"}' : Mon Aug 15 2016 11:26:05 GMT-0500 (CDT)
SocketService.js:91 Receiving: {"objectList":[{"objnam":"H66AE","params":{"OBJTYP":"HEATER","SNAME":"S2","STATUS":"OFF","SUBTYP":"SOLAR","PERMIT":"PERMIT","TIMOUT":"TIMOUT","READY":"READY","HTMODE":"SOLAR","SHOMNU":"hrtcoe","COOL":"ON","COMUART":"COMUART","BODY":"B6177","HNAME":"Solar Heater 1","START":"5","STOP":"2","HEATING":"OFF"}}],"command":"NotifyList","messageID":"79e279d7-5d5e-4ecd-b129-8601c5ecca9d","response":"200","timeNow":"947031207","timeSince":"947031112"}: Mon Aug 15 2016 11:26:05 GMT-0500 (CDT)

**Comment by acarnagey** [comment]
This is fixed, verified on dev2.  Pushed some changes on my side as well to dev.
